### ISSUE TITLE
Violating breaking change contract

### ISSUE THREAD

**Issue by creativecreatorormaybenot** [dart-code-checker/dart-code-metrics#322]
You are not versioning breaking changes correctly:

>warning - lines-of-executable-code deprecated option. This option will be removed in 3.3 version 

Any **breaking change** (like removing an option) requires a **major version bump** according to semantic versioning.

---

I noticed this because `3.1.0` actually violates this already IMO. You **deprecated** `lines-of-executable-code`. A deprecation is **usually** not a breaking change. However, in this case (as this is an analysis package), it will make analysis fail.

CI now fails with the warning included above (which is breaking).

**Comment by creativecreatorormaybenot** [comment]
Therefore, you should bump to `4.0.0` instead.

**Comment by dkrutskikh** [comment]
@creativecreatorormaybenot shure, if we change public api - we bump major. But major version not save our customers whose global activate this package.

**Comment by creativecreatorormaybenot** [comment]
What do you mean by:

> whose global activate this package.

**Comment by dkrutskikh** [comment]
> What do you mean by:
> 
> > whose global activate this package.

If you use `dart pub global activate`

**Comment by incendial** [comment]
@creativecreatorormaybenot thanks for feedback, we discussed it earlier and chose 3.3, but seems that we were wrong. Will publish 4.0 instead.

> CI now fails with the warning included above (which is breaking).

I don't see any code that is intensionally failing a pipeline because of a deprecation. Could you share more info about this?

`dart pub global activate` will install the latest version of a package even if it was a major change, which will lead to failing CI.

**Comment by dkrutskikh** [comment]
@creativecreatorormaybenot can you provide more details why ci drop builds?

**Comment by creativecreatorormaybenot** [comment]
@incendial thanks for the information.

@dkrutskikh CI fails because `flutter analyze` (`dart analyze`) now fails because in our `analysis_options.yaml` file, we declared `lines-of-executable-code: 80`.

```yaml
dart_code_metrics:
  metrics:
    lines-of-executable-code: 80
```

This is of course easily fixable by removing it, but that requires an additional change. Also note that we are not using `dart pub global activate` for `dart_code_metrics` - only the default analysis.

**Comment by incendial** [comment]
@creativecreatorormaybenot I was able to reproduce this warning only once, so can you please share some additional info:
1. do you have a stable reproduce? 
2. for how long it takes for `flutter analyze` to complete? Right now I think that this warning may appear only for runs longer then 20s.

**Comment by creativecreatorormaybenot** [comment]
@incendial 

1. Yes, it always happens.
2. It appears right away. The CI is very slow but on my machine it appears instantly basically:

```
warning • lines-of-executable-code deprecated option. This option will be removed in 3.3 version • analysis_options.yaml:0:0 • dart-code-metrics
1 issue found. (ran in 345.5s)
```

Make sure that you have configured the `analysis_options` file correctly would be my only guess.

**Comment by incendial** [comment]
Thanks for sharing the info. Seems like when `flutter analyze` processes plugin output, it's not awaiting the results properly. 
```
plants_care_app % flutter analyze 
Analyzing plants_care_app...                                            

warning • lines-of-executable-code deprecated option. This option will be removed in 3.3 version • analysis_options.yaml:0:0 •
       dart-code-metrics

1 issue found. (ran in 37.2s)
plants_care_app % flutter analyze
Analyzing plants_care_app...                                            
No issues found! (ran in 11.1s)
plants_care_app % flutter analyze
Analyzing plants_care_app...                                            
No issues found! (ran in 10.6s)
```
Only one with `37.2s` showed the warning.

We published version 3.2.2 where this check is disabled. Please let us know if you face this issue again.
